kubespawner/spawner.py

```diff
@@ -1544,9 +1637,50 @@ async def get_pod_manifest(self):
             pod_anti_affinity_preferred=self.pod_anti_affinity_preferred,
             pod_anti_affinity_required=self.pod_anti_affinity_required,
             priority_class_name=self.priority_class_name,
+            ssl_secret_name=self.secret_name,
+            ssl_secret_mount_path=self.secret_mount_path,
             logger=self.log,
         )
 
+    def get_secret_manifest(self, owner_reference):
+        """
+        Make a secret manifest that contains the ssl certificates.
+        """
+
+        labels = self._build_common_labels(self._expand_all(self.extra_labels))
+        annotations = self._build_common_annotations(
+            self._expand_all(self.extra_annotations)
+        )
+
+        return make_secret(
+            name=self.secret_name,
+            username=self.user.name,
+            cert_paths=self.cert_paths,
+            hub_ca=self.internal_trust_bundles['hub-ca'],
+            owner_references=[owner_reference],
+            labels=labels,
+            annotations=annotations,
+        )
+
+    def get_service_manifest(self, owner_reference):
+        """
+        Make a service manifest for dns.
+        """
+
+        labels = self._build_common_labels(self._expand_all(self.extra_labels))
+        annotations = self._build_common_annotations(
+            self._expand_all(self.extra_annotations)
+        )
+
+        return make_service(
+            name=self.pod_name,
```

Comment:
In the future: maybe add validation (or hash+truncation) to ensure valid service names. Check different requirements for service vs pod. Not for this PR, though.